library/src/main/kotlin/me/saket/filesize/FileSize.kt

```diff
@@ -76,38 +84,61 @@ class FileSize(val bytes: Long): Comparable<FileSize> {
     @PublishedApi internal const val BytesPerMb: Long = 1_000L * BytesPerKb
     @PublishedApi internal const val BytesPerGb: Long = 1_000L * BytesPerMb
 
-    inline val Int.bytes: FileSize
-      get() = FileSize(bytes = this.toLong())
-
-    inline val Long.bytes: FileSize
-      get() = FileSize(bytes = this)
-
-    inline val Int.kilobytes: FileSize
-      get() = FileSize(bytes = this * BytesPerKb)
-
-    inline val Double.kilobytes: FileSize
-      get() = FileSize(bytes = (this * BytesPerKb).toLong())
-
-    inline val Long.kilobytes: FileSize
-      get() = FileSize(bytes = (this * BytesPerKb))
-
-    inline val Int.megabytes: FileSize
-      get() = FileSize(bytes = this * BytesPerMb)
-
-    inline val Double.megabytes: FileSize
-      get() = FileSize(bytes = (this * BytesPerMb).toLong())
-
-    inline val Long.megabytes: FileSize
-      get() = FileSize(bytes = (this * BytesPerMb))
-
-    inline val Int.gigabytes: FileSize
-      get() = FileSize(bytes = this * BytesPerGb)
-
-    inline val Double.gigabytes: FileSize
-      get() = FileSize(bytes = (this * BytesPerGb).toLong())
-
-    inline val Long.gigabytes: FileSize
-      get() = FileSize(bytes = (this * BytesPerGb))
+    inline val Number.bytes: FileSize
+      get() {
+        val result = when (this) {
+          is Byte -> this.toLong()
+          is Short -> this.toLong()
+          is Int -> this.toLong()
+          is Long -> this
+          is Float -> this.toLong()
+          is Double -> this.toLong()
```

Comment:
`Double#bytes` currently produces a compilation error:

https://github.com/saket/file-size/blob/893a7c62a75dce28b7c8e446d78e916923d680bf/library/src/main/kotlin/me/saket/filesize/FileSize.kt#L112-L119

how can `Number#bytes` handle this?

Reply:
It's explicit.

<img width="1181" alt="image" src="https://github.com/saket/file-size/assets/10363352/f9fcc45a-cb5a-4ea7-9de2-c6d6e7c975c9">

Reply:
LGTM!